src/App.tsx

```diff
@@ -59,18 +59,88 @@ function App() {
   const [selectedAccounts, setSelectedAccounts] = useState<
     Record<string, `${string}:${string}:${string}`>
   >({});
+  const [walletSessionChangedHistory, setWalletSessionChangedHistory] =
+    useState<{ timestamp: number; data: any }[]>([]);
+
+  const handleSessionChangedNotification = (notification: any) => {
+    setWalletSessionChangedHistory((prev) => [
+      { timestamp: Date.now(), data: notification },
+      ...prev,
+    ]);
+
+    if (notification.params?.sessionScopes) {
+      setCreateSessionResult({
+        sessionScopes: notification.params.sessionScopes,
+      });
+      setGetSessionResult({
+        sessionScopes: notification.params.sessionScopes,
+      });
```

Comment:
not certain that we want the wallet_sessionChanged event to replace the results in these sections too. If the user changes their permissions via the wallet, it would overwrite what was previously in these areas. Not what I would personally expect to happen

Reply:
Cleaned up this logic here so that the "connected" state for accounts, chains, scopes cards now has a separate "currentSession" state separate from this state. Also made all of the method calls push to a history for better UX